spring-integration-file/src/main/java/org/springframework/integration/file/support/FileUtils.java

```diff
@@ -38,19 +40,29 @@
 	 * Remove entries from the array if the predicate returns true for an element.
 	 * @param fileArray the array.
 	 * @param predicate the predicate.
+	 * @param comparator an optional comparator to sort the results.
 	 * @param <F> the file type.
 	 * @return the array of remaining elements.
 	 * @since 5.0.7
 	 */
 	@SuppressWarnings("unchecked")
-	public static <F> F[] purgeUnwantedElements(F[] fileArray, Predicate<F> predicate) {
+	public static <F> F[] purgeUnwantedElements(F[] fileArray, Predicate<F> predicate,
+			@Nullable Comparator<F> comparator) {
 		if (ObjectUtils.isEmpty(fileArray)) {
 			return fileArray;
 		}
 		else {
-			return Arrays.stream(fileArray)
-					.filter(predicate.negate())
-					.toArray(size -> (F[]) Array.newInstance(fileArray[0].getClass(), size));
+			if (comparator == null) {
+				return Arrays.stream(fileArray)
+						.filter(predicate.negate())
+						.toArray(size -> (F[]) Array.newInstance(fileArray[0].getClass(), size));
+			}
+			else {
+				return Arrays.stream(fileArray)
+						.filter(predicate.negate())
+						.sorted(comparator)
```

Comment:
Can't we extract only this part of stream to the `if...else`?
I mean that there are enough copy/pasted code that it would be hard to support in both branches when it becomes much complicated eventually.

Although we can extract it at that point then, not now...

Reply:
Yeah - my reasoning here was that temporary storage in local variables is avoided this way.